turbopack/crates/turbopack-core/src/traced_asset.rs

```diff
@@ -32,9 +39,45 @@ impl OutputAsset for TracedAsset {
 
     #[turbo_tasks::function]
     async fn references(&self) -> Result<Vc<OutputAssets>> {
+        static SHARP_BINARY_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+            Regex::new("node_modules/@img/sharp-\\w+-\\w+/lib/sharp-(\\w+-\\w+).node$").unwrap()
+        });
+        let module_path = self.module.ident().path().await?;
+
+        // so sharp does require("@img/sharp-${arch}/sharp.node")
+        // @img/sharp-darwin-arm64/lib/sharp-darwin-arm64.node uses the dylib loading mechanism to
```

Comment:
nit: leave a reference to the vercel/nft implementation
https://github.com/vercel/nft/blob/7e915aa02073ec57dc0d6528c419a4baa0f03d40/src/utils/special-cases.ts#L151-L181